src/client/mapblock_mesh.cpp

```diff
@@ -422,14 +427,12 @@ void getNodeTile(MapNode mn, const v3s16 &p, const v3s16 &dir, MeshMakeData *dat
 static void applyTileColor(PreMeshBuffer &pmb)
 {
 	video::SColor tc = pmb.layer.color;
-	if (tc == video::SColor(0xFFFFFFFF))
-		return;
```

Comment:
This shouldn't be ditched.

Reply:
I think should be.`0xFFFFFFFF` means no the tile color and since it's multiplied to the texture one it will do the all nodes black. In the master it is added instead multiplying.

Reply:
No, the issue are the 0's returned from encode light, I'm pretty sure if you apply both my points here together it'll work.